lib/entryFromResponse.js

```diff
@@ -51,6 +51,13 @@ module.exports = function(entry, response, page, options) {
           })
       );
     }
+    // Merging and de-duplicating because extraResponse headers contain the original response headers as well
+    if (entry.extraResponseInfo.headers) {
+      const mergedHeaders = new Set(
+        [...headers, ...entry.extraResponseInfo.headers].map(JSON.stringify)
+      );
+      headers = Array.from(mergedHeaders, JSON.parse);
+    }
```

Comment:
This is the only section that really jumped out to me (not a maintainer).

Is it possible that `extraResponseInfo.headers` _always_ includes all of `headers`? if so, we can avoid all of this computation and just use that, yes?

I don't think we can rely on unique header names, so this solution certainly works.
Per [RFC7230 3.2.2](https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.2):
>   A sender MUST NOT generate multiple header fields with the same field
   name in a message unless either the entire field value for that
   header field is defined as a comma-separated list [i.e., #(values)]
   or the header field is a well-known exception (as noted below).
> 
>   A recipient MAY combine multiple header fields with the same field
   name into one "field-name: field-value" pair, without changing the
   semantics of the message, by appending each subsequent field value to
   the combined field value in order, separated by a comma.  The order
   in which header fields with the same field name are received is
   therefore significant to the interpretation of the combined field
   value; a proxy MUST NOT change the order of these field values when
   forwarding a message.

The only thing I could think of that might work better (and here we'd need some performance tests) is to use a Map instead.:

```js
const list = [
  {name: 'a', value: 1},
  {name: 'a', value: 2},
  {name: 'b', value: 1},
  {name: 'b', value: 1},
]

function makeUniqueSet(list) {
  const map = new Map()
  list.forEach(l => map.set(`${l.name}-${l.value}`, l))
  const uniqueSet = [...map.values()]
  return uniqueSet;
}
```

[Example Repl.it](https://replit.com/join/wigwjksxij-stephencweiss)

Reply:
Thanks Stephen! Made use of your suggestion and updated the PR. Will leave this comment open though to see if @soulgalore thinks this step is even necessary.